dataservice/api/common/custom_fields.py

```diff
@@ -0,0 +1,31 @@
+from dateutil import parser
+
+from marshmallow import (
+    fields,
+    ValidationError
+)
+
+
+class DateOrDatetime(fields.DateTime):
+    """
+    Custom field that represents a date/datetime field
+    Provide flexible deserialization by allowing values
+    that are either Date or DateTime objects
+
+    * NOTE - Marshmallow's _deserialize method assumes an iso
```

Comment:
See https://github.com/marshmallow-code/marshmallow/blob/3fd59becd91ac95617b0c06ecee2389dfe88dfda/marshmallow/fields.py#L965

This method might have a bug, looks like the `elif utils.dateutil_available` block can never execute?

Reply:
Looks like `elif self.dateformat` is always true? 
You should submit an issue and offer  a PR here:
https://github.com/marshmallow-code/marshmallow/pulls

Reply:
Yea, I wanted to but didn't have time today

Reply:
Aww darn :( I don't know how I missed this but someone already got to it: 

https://github.com/marshmallow-code/marshmallow/pull/619

Reply:
Should create an issue in our repo to make sure we track this.